chapters/how_does_opencl-opengl_interop.md

```diff
@@ -0,0 +1,47 @@
+# OpenCL-OpenGL interop
+
+Both OpenCL and OpenGL have specific extensions targeting resource sharing and synchronizing between the two runtimes. Doing so one may omit fetching data from the device, only to send it immediately back resulting in significant performane gains. Because the way the two APIs work, there are few thing to keep in mind when designing applications that intend interoperating.
+
+## How is it different than using OpenGL compute shaders?
+
+OpenGL compute shaders are slightly more restricted than OpenCL compute kernels. This is also reflected in the duality of the intermediate formats they can be compiled to. When using SPIR-V as an intermediate representation (IR), compute shaders are compiled to the graphics flavor of SIPR-V, which must exhibit structured control flow and must not use pointer arithmetic. These two cannot arise when using GLSL or other traditional shading languages. OpenCL C, being a C-derivate is far more liberal in the expressable language constructs than shading languages and as such requires a more feature complete intermediate representation, the so called compute flavor of SPIR-V. Different compiler infrastructure is required behind the scenes to process these two types of workloads, irrespective of ingesting IR or compiling from source.
```

Comment:
I think it's a bit of a distraction to discuss the IL formats here.  I wonder if we can make this a bit simpler.  To me, the key points are:

1. OpenCL kernels are (generally) more capable than GLSL compute shaders.
2. OpenCL kernels use a familiar C and C++ syntax compared to GLSL compute shaders.

For (2) we could link to the [C++ for OpenCL](https://github.com/KhronosGroup/OpenCL-Guide/blob/main/chapters/cpp_for_opencl.md) section of this guide as an example.